src/main/java/org/codehaus/plexus/components/io/fileselectors/IncludeExcludeFileSelector.java

```diff
@@ -24,11 +24,13 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
+import javax.inject.Named;
 
 /**
  * This file selector uses a set of patterns for including/excluding
  * files.
  */
+@Named( IncludeExcludeFileSelector.ROLE_HINT )
```

Comment:
This was singleton.

Reply:
This was wrong then: this class is stateful, it cannot be singleton.